server/routerlicious/packages/services-core/src/checkpointService.ts

```diff
@@ -31,62 +31,102 @@ export class CheckpointService implements ICheckpointService {
 		service: DocumentLambda,
 		checkpoint: IScribe | IDeliState,
 		isLocal: boolean = false,
+		markAsCorrupt: boolean = false,
+	) {
+		// services may not be able to process documents in corrupted state
+		// so we write to local and global databases when marking document as corrupt
+		if (markAsCorrupt) {
```

Comment:
Are we able to skip calling this if documents read already corruppted from caller?

Reply:
There are 2 cases for corrupted documents. 
- When a scribe lambda is created for a document that is marked as corrupt in the database, we create a NoOpLambda, so we will not run into this situation.
- When a scribe lambda has already been created and the error is thrown that causes the document to be marked as corrupted, `this.isDocumentCorrupted` is set to true, and here `markAsCorrupt` is true so we checkpoint to the local and global db to persist this. Then, the document is also marked as corrupt in memory, and we will do no more processing, and therefore will not try to checkpoint again. So I don't think we need a change to skip this. 
- 
@pradeepvairamani, can you confirm that the lambda does no more processing in the second situation here?

Reply:
In testing, I found that the document is no longer processed once marked as corrupt